design/handle-backup-of-volumes-by-resources-filters.md

```diff
@@ -0,0 +1,223 @@
+
+Taking select PV as an example, if users want to backup PV with name nginx-logs, the `groupResource` could be "/persistentvolumes" in which the group should be empty, the `namespacedNames` could be "/nginx-logs" in which the namespace should be empty.
+
+#### storage
+In the storage part, we defined `Volumes Attributes Selector` to filter resources.
+ 
+The storage part defined rules including `pv` and `volume`, which correspond to `Kopia, Restic` and `Volume snapshot`.
+ 
+A filter in storage with a specific key and empty value, which means the value matches any value. For example, if the `storage.pv.exclude.persistentVolumeSource.nfs` is `{}` it means if `NFS` is used as `persistentVolumeSource` in Persistent Volume will be skipped no matter what the NFS server or NFS Path is,  
+ 
+A filter may have multiple values, all the values are concatenated by commas. For example, the `storage.pv.include.storageClassName` is `gp2, ebs-sc` which means Persistent Volume with gp2 or ebs-sc storage class both will be back up.
+ 
+The size of each single filter value should limit to 256 bytes in case of an unfriendly long variable assignment.
+ 
+If user defined pv filter rules but used Kopia or Restic to do a backup, the backup will fail in validating the resource filter configuration. Same as the situation if using defined volume filter rules but using CSI or plugins to take volume snapshots.
+ 
+For capacity in `pv` or size in `volume`, the value should include the lower value and upper value concatenated by commas. And it has several combinations below:
+- "0,5Gi" or "0Gi,5Gi" which means capacity or size matches from 0 to 5Gi, including value 0 and value 5Gi
+- ",5Gi" which is equal to "0,5Gi"
+- "5Gi," which means capacity or size matches larger than 5Gi, including value 5Gi
+- "5Gi" which is not supported and will be failed in validating configuration.
+ 
+### Filter Reference
+Currently, resources filters are defined in `BackupSpec` struct, it will be more and more bloated with adding more and more filters which makes the size of `Backup` CR bigger and bigger, so we want to store the resources rules in configmap, and `Backup` CRD reference to current configmap.
+
+the `configmap` would be like this:
+```
+apiVersion: v1
+data:
+ filter.json: |-
+   {
+       "resources": {
+           "include": {
+              ...
+           },
+           "exclude": {
+              ...
+           }
+       },
+       "storage": {
+           "pv": {
+               "include": {
+                  ...
+               },
+               "exclude": {
+                  ...
+               }
+           },
+           "volume": {
+               "include": {
+                  ...
+               },
+               "exclude": {
+                  ...
+               }
+           }
+       }
+   }
+kind: ConfigMap
+metadata:
+ creationTimestamp: "2023-01-16T14:08:12Z"
+ name: cm-backup01
+ namespace: default
```

Comment:
Shall we keep everything in velero namespace

Reply:
yes, the value is incorrect, I'll modify it to velero.

Reply:
yes, it should be in velero namespace